internal/flavors/asset_inventory.go

```diff
@@ -49,13 +53,26 @@ func newAssetInventoryFromCfg(b *beat.Beat, cfg *config.Config) (*assetInventory
 	logger := logp.NewLogger("asset_inventory")
 	ctx, cancel := context.WithCancel(context.Background())
 
-	logger.Info("Creating AWS AssetInventory")
-
-	awsFetchers, err := initAwsFetchers(ctx, cfg, logger)
+	var fetchers []inventory.AssetFetcher
+	var err error
+
+	switch cfg.AssetInventoryProvider {
```

Comment:
I do like how on cspm this is extracted to an strategy file. It's clean.

You don't need to implement it, I'm just pointing how cspm does. Maybe it influences you.

https://github.com/elastic/cloudbeat/blob/main/internal/flavors/benchmark/strategy.go

Reply:
Makes sense, especially if we go with the `Type` route.